src/hotspot/share/opto/subnode.cpp

```diff
@@ -185,6 +185,17 @@ Node *SubINode::Ideal(PhaseGVN *phase, bool can_reshape){
       return new AddINode(in1, phase->intcon(-i->get_con()));
   }
 
+  // Convert "c0 - (x + c1)" into "(c0 - c1) - x"
+  if (in2->Opcode() == Op_AddI
+      && phase->type(in1)->isa_int() != NULL
+      && phase->type(in1)->isa_int()->is_con()
+      && phase->type(in2->in(2))->isa_int() != NULL
+      && phase->type(in2->in(2))->isa_int()->is_con()) {
+      jint c0 = phase->type(in1)->isa_int()->get_con();
+      jint c1 = phase->type(in2->in(2))->isa_int()->get_con();
+      return new SubINode(phase->intcon(java_subtract(c0, c1)), in2->in(1));
+  }
```

Comment:
Should this code be checking for counted loops like below (see ok_to_convert)?

Reply:
Makes sense. Added checking for "x + c1".